src/world_interface/kinematic_common_interface.cpp

```diff
@@ -77,6 +86,106 @@ double setCmdVel(double dtheta, double dx) {
 	return maxAbsPWM > 1 ? maxAbsPWM : 1.0;
 }
 
+double setTankCmdVel(double left, double right) {
+	if (Globals::E_STOP && (left != 0 || right != 0)) {
+		return 0;
+	}
+
+	if (!std::get<1>(Globals::driveMode)) {
+		for (int i = 0; i < 4; i++) {
+			if (std::abs(robot::getMotorPos(Constants::WHEEL_IDS[i]) -
+						 Constants::WHEEL_ROTS.at(DriveMode::Normal)[i]) <
+				Constants::STEER_EPSILON)
+				return 0;
+		}
+	}
+
+	wheelvel_t wheelVels = {left, right};
+	double lPWM = wheelVels.lVel / Constants::MAX_WHEEL_VEL;
+	double rPWM = wheelVels.rVel / Constants::MAX_WHEEL_VEL;
+	double maxAbsPWM = std::max(std::abs(lPWM), std::abs(rPWM));
+	if (maxAbsPWM > 1) {
+		lPWM /= maxAbsPWM;
+		rPWM /= maxAbsPWM;
+	}
+
+	setCmdVelToIntegrate(wheelVels);
+	setMotorPower(motorid_t::frontLeftWheel, lPWM);
+	setMotorPower(motorid_t::rearLeftWheel, lPWM);
+	setMotorPower(motorid_t::frontRightWheel, rPWM);
+	setMotorPower(motorid_t::rearRightWheel, rPWM);
+
+	return maxAbsPWM > 1 ? maxAbsPWM : 1.0;
+}
+
+double setTurnInPlaceCmdVel(double dtheta) {
+	if (Globals::E_STOP && (dtheta != 0)) {
+		return 0;
+	}
+
+	if (!std::get<1>(Globals::driveMode)) {
+		for (int i = 0; i < 4; i++) {
+			if (std::abs(robot::getMotorPos(Constants::WHEEL_IDS[i]) -
+						 Constants::WHEEL_ROTS.at(DriveMode::TurnInPlace)[i]) <
+				Constants::STEER_EPSILON)
+				return 0;
+		}
+	}
```

Comment:
Is there an easy way to factor this out into a little function? It's used in 3 places throughout. No big deal if not